core/shared/src/main/scala/org/virtuslab/yaml/YamlEncoder.scala

```diff
@@ -18,6 +18,11 @@ object YamlEncoder extends YamlEncoderCrossCompanionCompat {
   implicit def forBoolean: YamlEncoder[Boolean] = v => Node.ScalarNode(v.toString)
   implicit def forString: YamlEncoder[String]   = v => Node.ScalarNode(v)
 
+  implicit def forOption[T](implicit encoder: YamlEncoder[T]): YamlEncoder[Option[T]] = {
+    case Some(t) => encoder.asNode(t)
+    case None    => Node.ScalarNode("", Tag.nullTag)
```

Comment:
is there some reason to encode null as scalar node with null tag? I know this should be question to myself, but just thinking loudly

Reply:
There are no Null nodes and I didn't check if `null` can be passed as a value to Scalar node. I wouldn't expect it to be tbh.

Reply:
`Node.ScalarNode("", Tag.nullTag)` seems like a hack to represent something more complex that model was designed for :D It isn't obvious for me, but maybe it is ok because it is how spec defines it - https://yaml.org/spec/1.2.2/#10211-null.

Reply:
ehh, I now saw the how `Tag` looks like. I think it's ok as it is tag which determines how value of scalar should be interpreted